scripts/core/queue.yml

```diff
@@ -332,3 +332,49 @@ params:
 returns:
     - $X_RESULT_ERROR_INVALID_QUEUE
     - $X_RESULT_ERROR_OUT_OF_HOST_MEMORY
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Get the suggested local work-item number from runtime implementation."
+class: $xQueue
+name: GetSuggestedLocalWorkSize
+ordinal: "0"
+details:
+    - "pLocalWorkSize can be omitted in urEnqueueKernelLaunch(), but beside from"
+    - "OpenCL, LocalWorkSize will need to be calculated or guessed before enqueue"
+    - "the kernel. This function will get the LocalWorkSize value used when enqueueing"
+    - "the kernel."
+    - "TODO: find a better place for this function"
+params:
+    - type: $x_queue_handle_t
+      name: hQueue
+      desc: |
+            [in] handle of the queue object
```

Comment:
Is the queue required here or is it being used to access the device?

If the queue's internal state is not required, we should just pass in a device instead.

Reply:
@npmiller for CUDA/HIP would having access to the device rather than the queue provide sufficent information to implement this?

Reply:
> Is the queue required here or is it being used to access the device?

For OpenCL implementation, the queue is required(because its API requires the queue).